src/ProtocolFeeController.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2024 PancakeSwap
+pragma solidity 0.8.26;
+
+import {Ownable} from "./base/Ownable.sol";
+import {PoolKey} from "./types/PoolKey.sol";
+import {PoolId} from "./types/PoolId.sol";
+import {Currency} from "./types/Currency.sol";
+import {LPFeeLibrary} from "./libraries/LPFeeLibrary.sol";
+import {IProtocolFeeController} from "./interfaces/IProtocolFeeController.sol";
+import {IProtocolFees} from "./interfaces/IProtocolFees.sol";
+import {ProtocolFeeLibrary} from "./libraries/ProtocolFeeLibrary.sol";
+
+/// @notice ProtocolFeeController for both Pool type
```

Comment:
It seems this contract can be reused by both pool types, is it ? (we can even only deploy one controller for two pools)

Reply:
So basically we have 3 opitions here:

1. write two different fee controller for different pool types
2. reuse the same logic for both pool types but deploy separately (currently selected)
3. reuse the same logic for both pool types and deploy only one contract

Reply:
i vote 2

Reply:
no strong opinion as well

Reply:
then let's go with vote2 (keep currently implementation

Reply:
marking this as resolved